### ISSUE TITLE
Save fails on external storage with "insuficient storage" error in 3.1.2

### ISSUE THREAD

**Issue by flakeparadigm** [nextcloud/notes#455]
**Describe the bug**
After upgrading to v3.1.2, the notes that I have stored in an SMB-backed External Storage in my account cannot be saved after editing or creating a new one. Attempting to do so always results in a notification that "Saving the note has failed due to insufficient storage."

**To Reproduce**
1. Install or upgrade to Notes v3.1.2
2. Go to Notes app
3. Set the storage to a location created by the "External storage support" app, with plenty of space available
4. Attempt to create a new note, or edit an existing note
5. See error "Saving the note has failed due to insufficient storage."

**Expected behavior**
Notes save without issue on External Storage when there is indeed sufficient space to save them.

**Screenshots**
<img width="441" alt="Screenshot 2020-02-05 at 09 13 16" src="https://user-images.githubusercontent.com/860852/73792106-c40e7f00-47f7-11ea-876f-a54a664394c5.png">

**Server (please complete the following information):**
- Notes app version: 3.1.2
- Nextcloud version: 18.0.0 (docker: nextcloud/fpm)
- OS: Linux/docker
- Web server: Nginx (docker: nginx/mainline)
- PHP version: 7.3.14
- Database: mariadb 10.4.12

**Client (please complete the following information):**
 - Browser (incl. version): Firefox 73.0b12
 - OS: macOS 10.14.6

**Log files**
Cannot find any logs output when this error occurs in either:
- /var/www/html/data/nextcloud.log (in docker containers)
- logs for the docker containers (nginx and fpm)

Browser console:
```
Error: "Request failed with status code 507" [notes.js:24:319459]
    exports [/custom_apps/notes/js/notes.js?v=dbc4884e-10:8]
    exports [/custom_apps/notes/js/notes.js?v=dbc4884e-10:14]
    onreadystatechange [/custom_apps/notes/js/notes.js?v=dbc4884e-10:8]
    ...
    [snipped for brevity, likely not the most helpful because it's minified/compiled]
```

**Comment by flakeparadigm** [comment]
As a temporary fix so I can keep using Notes, I have moved all of my notes from my external storage into the actual Nextcloud storage for my user. While this isn't optimal/desired for me, it's at least a sufficient temporary workaround.

**Comment by korelstar** [comment]
Thanks for your bug report! I was able to reproduce the bug and will try to provide a fix soon.

**Comment by flakeparadigm** [comment]
Wow, thanks for the quick fix on that! Works properly for me as well after updating to 3.1.3 👍

**Comment by faulteh** [comment]
HELP I upgraded to 3.1.3 this morning within the Apps interface and now I cannot create any new notes. It says cannot create new note, see JS Console for details. The Notes REST request is making a 500 error. The PHP log in Nextcloud is showing this:

I am running my notes on 'internal' storage (ie, not a SMB share as this issue is supposed to fix) but it looks like this issue's fix has broken my install here. I am running Nextcloud 17.0.3 on Ubuntu bionic with PHP 7.2.24 (Apache2)

I tried to copy in 3.1.1 back into my Nextcloud from the tarball but now Nextcloud just asks me to upgrade the Notes app straight away and it goes back to 3.1.3.

{"reqId":"469gXhO9oe56aVjGdN4B","level":3,"time":"2020-02-06T03:32:57+00:00","remoteAddr":"10.100.200.9","user":"faulteh","app":"index","method":"POST","url":"/apps/notes/notes","message":{"Exception":"Exception","Message":"Argument 1 passed to OCA\\Notes\\Service\\NoteUtil::ensureSufficientStorage() must be an instance of OCP\\Files\\File, null given, called in /home/www/oc.scriptforge.org/apps/notes/lib/Service/NotesService.php on line 129","Code":0,"Trace":[{"file":"/home/www/oc.scriptforge.org/lib/private/AppFramework/App.php","line":126,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Notes\\Controller\\NotesController"},"create"]},{"file":"/home/www/oc.scriptforge.org/lib/private/AppFramework/Routing/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Notes\\Controller\\NotesController","create",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"notes.notes.create"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"notes.notes.create"}]},{"file":"/home/www/oc.scriptforge.org/lib/private/Route/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"notes.notes.create"}]},{"file":"/home/www/oc.scriptforge.org/lib/base.php","line":997,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/notes/notes"]},{"file":"/home/www/oc.scriptforge.org/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/home/www/oc.scriptforge.org/lib/private/AppFramework/Http/Dispatcher.php","Line":109,"Previous":{"Exception":"TypeError","Message":"Argument 1 passed to OCA\\Notes\\Service\\NoteUtil::ensureSufficientStorage() must be an instance of OCP\\Files\\File, null given, called in /home/www/oc.scriptforge.org/apps/notes/lib/Service/NotesService.php on line 129","Code":0,"Trace":[{"file":"/home/www/oc.scriptforge.org/apps/notes/lib/Service/NotesService.php","line":129,"function":"ensureSufficientStorage","class":"OCA\\Notes\\Service\\NoteUtil","type":"->","args":[null,1]},{"file":"/home/www/oc.scriptforge.org/apps/notes/lib/Controller/NotesController.php","line":129,"function":"create","class":"OCA\\Notes\\Service\\NotesService","type":"->","args":["faulteh"]},{"file":"/home/www/oc.scriptforge.org/lib/private/AppFramework/Http/Dispatcher.php","line":170,"function":"create","class":"OCA\\Notes\\Controller\\NotesController","type":"->","args":["",null]},{"file":"/home/www/oc.scriptforge.org/lib/private/AppFramework/Http/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Notes\\Controller\\NotesController"},"create"]},{"file":"/home/www/oc.scriptforge.org/lib/private/AppFramework/App.php","line":126,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Notes\\Controller\\NotesController"},"create"]},{"file":"/home/www/oc.scriptforge.org/lib/private/AppFramework/Routing/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Notes\\Controller\\NotesController","create",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"notes.notes.create"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"notes.notes.create"}]},{"file":"/home/www/oc.scriptforge.org/lib/private/Route/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"notes.notes.create"}]},{"file":"/home/www/oc.scriptforge.org/lib/base.php","line":997,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/notes/notes"]},{"file":"/home/www/oc.scriptforge.org/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/home/www/oc.scriptforge.org/apps/notes/lib/Service/NoteUtil.php","Line":226},"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.87 Safari/537.36","version":"17.0.3.1","id":"5e3b89fcb4e4b"}

**Comment by faulteh** [comment]
$file is not assigned by line 129 when the exception is thrown.
It looks like the line in lib/Service/NotesService.php on line 129

$this->noteUtil->ensureSufficientStorage($file, 1);

needs to be moved to after where the $file variable is assigned, to line 135

Unfortunately I'm not on a machine with git on it to submit a pull request.

**Comment by korelstar** [comment]
Ouh, sorry. I think I was too fast with that release. I just released version 3.1.4 which fixes that regression bug.

**Comment by cubanpit** [comment]
I think there is still something wrong, with Notes 3.1.5, Nextcloud 15.0.14, PHP 7, I get the following error every time my smartphone Notes app tries to sync. 
```text
[index] Error: TypeError: Return value of OCA\Notes\Service\NoteUtil::ensureSufficientStorage() must be an instance of OCA\Notes\Service\void, none returned at <<closure>>

0. /var/www/nextcloud/apps/notes/lib/Service/NotesService.php line 129
   ensureSufficientStorage(OC\Files\Node\Folder {}, 1)
1. /var/www/nextcloud/apps/notes/lib/Controller/NotesApiController.php line 141
   create("cubanpit")
2. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 166
   create("How to make a g ...  ", "", 1582198031, false)
3. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 99
   executeController(OCA\Notes\Contro ... {}, "create")
4. /var/www/nextcloud/lib/private/AppFramework/App.php line 118
   dispatch(OCA\Notes\Contro ... {}, "create")
5. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   main("OCA\\Notes\\Con ... r", "create", OC\AppFramework\ ... {}, {_route: "notes.notes_api.create"})
6. <<closure>>
   __invoke({_route: "notes.notes_api.create"})
7. /var/www/nextcloud/lib/private/Route/Router.php line 297
   call_user_func(OC\AppFramework\ ... {}, {_route: "notes.notes_api.create"})
8. /var/www/nextcloud/lib/base.php line 987
   match("/apps/notes/api/v0.2/notes")
9. /var/www/nextcloud/index.php line 42
   handleRequest()

POST /nextcloud/index.php/apps/notes/api/v0.2/notes?
from x.x.x.x by cubanpit at 2020-02-21T08:57:39+00:00
```
I post here because this is probably related to the previous issue, otherwise let me know and I will open a new one.

**Comment by korelstar** [comment]
@cubanpit Yep, that's the same issue. But this has been fixed in version 3.1.5, so I think your web server uses a cached version of the file `lib/Service/NotesService.php`. Please try to restart your web server and your PHP-FPM service (if applicable).

**Comment by cubanpit** [comment]
You are right, reloading the `php-fpm` service solved the problem, I should have checked before reporting. Thank you for your suggestion!
